### PR TITLE
PDL modules to bfp8 and imagenet normalization for DEMO

### PR DESCRIPTION
**Pull request by lzdravicTT**
### Problem description
PanopticDeepLab Demo was not using all modules in bfp8
PanotpicDeepLab Demo was producing low quality images

### What's changed
Changed all modules to be in bfp8
Added imagenet normalization in image preprocessing part of demo run
Changed interpolate in pytorch to do nearest too, so we reproduce sharp and pixelated edges, on pytorch side too, for now.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Blackhole Nightly](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-nightly-tests.yaml) CI with PCC tests passes
- [ ] [L2 Nightly](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml)

### CODE REVIEW

**Review by ianastasijevicTT — APPROVED** [review]
Approved assuming CI pipelines are green.

File: models/experimental/panoptic_deeplab/tt/tt_resnet.py [hunk 1]
```diff
@@ -25,21 +25,45 @@ def __init__(
         self,
         parameters,
         device: ttnn.MeshDevice,
-        dtype: ttnn.DataType = ttnn.bfloat16,
+        dtype: ttnn.DataType = ttnn.bfloat8_b,
+        layer_dtypes: dict = None,
```
Comment (mbezuljTT) [inline models/experimental/panoptic_deeplab/tt/tt_resnet.py]: this pattern with dtype and layer_dtypes make user unclear one is real way to control dtype. I suggest using Union[ttnn.DataType, dict[str, ttnn.DataType]] so it's clear that only one could be used.

File: models/experimental/panoptic_deeplab/tt/demo.py [hunk 2]
```diff
@@ -196,6 +198,11 @@ def preprocess_image(image_path: str, target_size: Tuple[int, int] = (512, 1024)
     # Convert to tensor and normalize to [0, 1]
     image_tensor = torch.from_numpy(image.transpose(2, 0, 1)).float() / 255.0
 
+    if use_imagenet_norm:
+        mean = torch.tensor([0.485, 0.456, 0.406]).view(3, 1, 1)
+        std = torch.tensor([0.229, 0.224, 0.225]).view(3, 1, 1)
+        image_tensor = (image_tensor - mean) / std
```
Comment (mbezuljTT) [inline models/experimental/panoptic_deeplab/tt/demo.py]: we want to run normalization on tensix.

File: models/experimental/panoptic_deeplab/reference/pytorch_aspp.py [hunk 3]
```diff
@@ -173,7 +173,7 @@ def forward(self, x):
 
         # Process pooling branch (convs[4]) with interpolation
         pool_out = self.convs[-1](x)
-        pool_out = F.interpolate(pool_out, size=size, mode="bilinear", align_corners=False).to(x.dtype)
+        pool_out = F.interpolate(pool_out, size=size, mode="nearest").to(x.dtype)
```
Comment (mbezuljTT) [inline models/experimental/panoptic_deeplab/reference/pytorch_aspp.py]: please extend pytorch reference so interpolation could be neearest or bilinear via model configuration.

**Review by mbezuljTT — CHANGES_REQUESTED** [review]

**Review by mbezuljTT — CHANGES_REQUESTED** [review]
minor changes

File: models/experimental/panoptic_deeplab/reference/pytorch_aspp.py (hunk 3 above)
Comment (lzdravicTT) [reply models/experimental/panoptic_deeplab/reference/pytorch_aspp.py]: Left it to bilinear since PCC was falling a lot when pytorch reference was using nearest, still don't know why

File: models/experimental/panoptic_deeplab/tt/demo.py [hunk 4]
```diff
@@ -202,6 +205,40 @@ def preprocess_image(image_path: str, target_size: Tuple[int, int] = (512, 1024)
     return image_tensor
 
 
+def apply_imagenet_norm_on_device(input_tensor: ttnn.Tensor, device: ttnn.Device) -> ttnn.Tensor:
```
Comment (mbezuljTT) [inline models/experimental/panoptic_deeplab/tt/demo.py]: per offline conversation, let's wrap normalization in the module so we have split init that happens once, and forward pass that we can call any number of times without of IO overheads.

File: models/experimental/panoptic_deeplab/tt/demo.py [hunk 5]
```diff
@@ -202,6 +152,63 @@ def preprocess_image(image_path: str, target_size: Tuple[int, int] = (512, 1024)
     return image_tensor
 
 
+class TtImageNetNormalization:
```
Comment (mbezuljTT) [inline models/experimental/panoptic_deeplab/tt/demo.py]: please move to separate module file under tt/

**Review by mbezuljTT — APPROVED** [review]
